<a href="https://colab.research.google.com/github/deba-roy/World_Bank_Global_Education_Analysis/blob/main/DEBARTHA_ROY_CHOWDHURY_World_Bank_Global_Education_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The World Bank EdStats All Indicator Query holds over 4,000 internationally comparable indicators that describe education access, progression, completion, literacy, teachers, population, and expenditures.The indicators cover the education cycle from pre-primary to vocational and tertiary education and also holds learning outcome data from international and regional learning assessments (e.g. PISA, TIMSS, PIRLS), equity data from household surveys, and projection/attainment data. </b>

## <b> Explore and analyze the data to identify variation of indicators across the globe, which countries are more alike and different. Feel free to add more extensive analyses and details.</b>

# World Bank Global Education Analysis
   **_Introduction:_** 
  
  The World Bank EdStats (Education Statistics) portal is a comprehensive data and analysis source for key topics in education such as access, completion, learning, expenditures, policy, and equity. Data sources include administrative country data from international learning assessments (PISA, TIMSS, PIRLS, PIAAC, and EGRA) and three regional learning assessments (SACMEQ, PASEC, LLECE); World Bank databases. It includes World Bank Education Projects Database classified by activities, components, and sub-sectors of all World Bank Education Projects since 1970.

## 1. Downloading and Importing the necessary files and tools

In [1]:
# Mounting drive 
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


### Importing the necessary libraries

In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

### Reading the given dataset from the csv files

In [5]:
df_stats_country_series = pd.read_csv('/content/drive/MyDrive/Almabetter/Modules/Python for Data science/Capstone Project/data')
df_stats_country = pd.read_csv('/content/drive/MyDrive/Almabetter/Modules/Python for Data science/Capstone Project/EdStatsCountry.csv')
df_stats_data = pd.read_csv('/content/drive/MyDrive/Almabetter/Modules/Python for Data science/Capstone Project/EdStatsData.csv')
df_stats_footnote = pd.read_csv('/content/drive/MyDrive/Almabetter/Modules/Python for Data science/Capstone Project/EdStatsFootNote.csv')
df_stats_series = pd.read_csv('/content/drive/MyDrive/Almabetter/Modules/Python for Data science/Capstone Project/EdStatsSeries.csv')

### Primary insights on each of the given datasets:
**_About the dataset:_** (source, contents, usefulness) 

*   df_stats_country_series(EdStatsCountry-Series.csv) - this list contains data sources and indicators respective to each countries. 
*   df_stats_country(EdStatsCountry.csv) - this list contains all countries and the other features related to the countries like region, income group specific to them.
*   df_stats_data (EdStatsData.csv) - it contains list of indicators(about 3665 of them), projections from the year 2020 to 2100 and measurement value for each indicator in the year range 1970 to 2017.
*   df_stats_footnote (EdStatsFootnote.csv) - it contains uncertainty and estimations for the years 
*   df_stats_series = (EdStatsSeries.csv) - containing list of the indicators and the definition of each of those.




## 2. Data Preparation and cleaning
  - load files using pandas
  - look for information about data and columns
  - fix any missing or incorrect values

#Analysis of EdStatsCountry-Series.csv

In [12]:
df_stats_country_series.head()

CountryCode         SeriesCode  \
0         ABW        SP.POP.TOTL   
1         ABW        SP.POP.GROW   
2         AFG        SP.POP.GROW   
3         AFG  NY.GDP.PCAP.PP.CD   
4         AFG        SP.POP.TOTL   

                                         DESCRIPTION  Unnamed: 3  
0  Data sources : United Nations World Population...         NaN  
1  Data sources: United Nations World Population ...         NaN  
2  Data sources: United Nations World Population ...         NaN  
3                 Estimates are based on regression.         NaN  
4  Data sources : United Nations World Population...         NaN

In [7]:
df_stats_country_series.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 613 entries, 0 to 612
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   CountryCode  613 non-null    object 
 1   SeriesCode   613 non-null    object 
 2   DESCRIPTION  613 non-null    object 
 3   Unnamed: 3   0 non-null      float64
dtypes: float64(1), object(3)
memory usage: 19.3+ KB


### This csv file contains unnamed: 3 that we have to remove

In [17]:
# code to remove the column unnamed: 3
New_Country_Series = df_stats_country_series.drop('Unnamed: 3',axis=1)
New_Country_Series


CountryCode         SeriesCode  \
0           ABW        SP.POP.TOTL   
1           ABW        SP.POP.GROW   
2           AFG        SP.POP.GROW   
3           AFG  NY.GDP.PCAP.PP.CD   
4           AFG        SP.POP.TOTL   
..          ...                ...   
608         ZAF        SP.POP.GROW   
609         ZMB        SP.POP.GROW   
610         ZMB        SP.POP.TOTL   
611         ZWE        SP.POP.TOTL   
612         ZWE        SP.POP.GROW   

                                           DESCRIPTION  
0    Data sources : United Nations World Population...  
1    Data sources: United Nations World Population ...  
2    Data sources: United Nations World Population ...  
3                   Estimates are based on regression.  
4    Data sources : United Nations World Population...  
..                                                 ...  
608  Data sources : Statistics South Africa, United...  
609  Data sources: United Nations World Population ...  
610  Data sources : United Nations World Population...  
611  Data sources : United Nations World Population...  
612  Data sources: United Nations World Population ...  

[613 rows x 3 columns]

In [18]:
New_Country_Series.describe()

CountryCode   SeriesCode  \
count          613          613   
unique         211           21   
top            MDA  SP.POP.TOTL   
freq            18          211   

                                              DESCRIPTION  
count                                                 613  
unique                                                 97  
top     Data sources : United Nations World Population...  
freq                                                  154

#Analysis of EdStatsCountry.csv

In [13]:
df_stats_country.head()

Country Code   Short Name   Table Name                     Long Name  \
0          ABW        Aruba        Aruba                         Aruba   
1          AFG  Afghanistan  Afghanistan  Islamic State of Afghanistan   
2          AGO       Angola       Angola   People's Republic of Angola   
3          ALB      Albania      Albania           Republic of Albania   
4          AND      Andorra      Andorra       Principality of Andorra   

  2-alpha code   Currency Unit  \
0           AW   Aruban florin   
1           AF  Afghan afghani   
2           AO  Angolan kwanza   
3           AL    Albanian lek   
4           AD            Euro   

                                       Special Notes  \
0  SNA data for 2000-2011 are updated from offici...   
1  Fiscal year end: March 20; reporting period fo...   
2  April 2013 database update: Based on IMF data,...   
3                                                NaN   
4                                                NaN   

                      Region          Income Group WB-2 code  ...  \
0  Latin America & Caribbean  High income: nonOECD        AW  ...   
1                 South Asia            Low income        AF  ...   
2         Sub-Saharan Africa   Upper middle income        AO  ...   
3      Europe & Central Asia   Upper middle income        AL  ...   
4      Europe & Central Asia  High income: nonOECD        AD  ...   

            IMF data dissemination standard  \
0                                       NaN   
1  General Data Dissemination System (GDDS)   
2  General Data Dissemination System (GDDS)   
3  General Data Dissemination System (GDDS)   
4                                       NaN   

                            Latest population census  \
0                                               2010   
1                                               1979   
2                                               1970   
3                                               2011   
4  2011. Population figures compiled from adminis...   

                             Latest household survey  \
0                                                NaN   
1  Multiple Indicator Cluster Survey (MICS), 2010/11   
2               Malaria Indicator Survey (MIS), 2011   
3       Demographic and Health Survey (DHS), 2008/09   
4                                                NaN   

   Source of most recent Income and expenditure data  \
0                                                NaN   
1            Integrated household survey (IHS), 2008   
2            Integrated household survey (IHS), 2008   
3  Living Standards Measurement Study Survey (LSM...   
4                                                NaN   

  Vital registration complete Latest agricultural census  \
0                         Yes                        NaN   
1                         NaN                    2013/14   
2                         NaN                       2015   
3                         Yes                       2012   
4                         Yes                        NaN   

  Latest industrial data Latest trade data Latest water withdrawal data  \
0                    NaN            2012.0                          NaN   
1                    NaN            2012.0                         2000   
2                    NaN               NaN                         2005   
3                 2010.0            2012.0                         2006   
4                    NaN            2006.0                          NaN   

  Unnamed: 31  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN  

[5 rows x 32 columns]

In [8]:
df_stats_country.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 241 entries, 0 to 240
Data columns (total 32 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   Country Code                                       241 non-null    object 
 1   Short Name                                         241 non-null    object 
 2   Table Name                                         241 non-null    object 
 3   Long Name                                          241 non-null    object 
 4   2-alpha code                                       238 non-null    object 
 5   Currency Unit                                      215 non-null    object 
 6   Special Notes                                      145 non-null    object 
 7   Region                                             214 non-null    object 
 8   Income Group                                       214 non-null    object 
 9   WB-2 code 

### This csv file contains unnamed: 31 that we have to remove

In [20]:
# code to remove the column unnamed: 31
New_Country = df_stats_country.drop('Unnamed: 31',axis=1)
New_Country

Country Code    Short Name    Table Name                     Long Name  \
0            ABW         Aruba         Aruba                         Aruba   
1            AFG   Afghanistan   Afghanistan  Islamic State of Afghanistan   
2            AGO        Angola        Angola   People's Republic of Angola   
3            ALB       Albania       Albania           Republic of Albania   
4            AND       Andorra       Andorra       Principality of Andorra   
..           ...           ...           ...                           ...   
236          XKX        Kosovo        Kosovo            Republic of Kosovo   
237          YEM         Yemen   Yemen, Rep.             Republic of Yemen   
238          ZAF  South Africa  South Africa      Republic of South Africa   
239          ZMB        Zambia        Zambia            Republic of Zambia   
240          ZWE      Zimbabwe      Zimbabwe          Republic of Zimbabwe   

    2-alpha code       Currency Unit  \
0             AW       Aruban florin   
1             AF      Afghan afghani   
2             AO      Angolan kwanza   
3             AL        Albanian lek   
4             AD                Euro   
..           ...                 ...   
236          NaN                Euro   
237           YE         Yemeni rial   
238           ZA  South African rand   
239           ZM  New Zambian kwacha   
240           ZW         U.S. dollar   

                                         Special Notes  \
0    SNA data for 2000-2011 are updated from offici...   
1    Fiscal year end: March 20; reporting period fo...   
2    April 2013 database update: Based on IMF data,...   
3                                                  NaN   
4                                                  NaN   
..                                                 ...   
236  Kosovo became a World Bank member on June 29, ...   
237  Based on official government statistics and In...   
238  Fiscal year end: March 31; reporting period fo...   
239  National accounts data have rebased to reflect...   
240  Fiscal year end: June 30; reporting period for...   

                         Region          Income Group WB-2 code  ...  \
0     Latin America & Caribbean  High income: nonOECD        AW  ...   
1                    South Asia            Low income        AF  ...   
2            Sub-Saharan Africa   Upper middle income        AO  ...   
3         Europe & Central Asia   Upper middle income        AL  ...   
4         Europe & Central Asia  High income: nonOECD        AD  ...   
..                          ...                   ...       ...  ...   
236       Europe & Central Asia   Lower middle income        KV  ...   
237  Middle East & North Africa   Lower middle income        RY  ...   
238          Sub-Saharan Africa   Upper middle income        ZA  ...   
239          Sub-Saharan Africa   Lower middle income        ZM  ...   
240          Sub-Saharan Africa            Low income        ZW  ...   

       Government Accounting concept  \
0                                NaN   
1    Consolidated central government   
2       Budgetary central government   
3       Budgetary central government   
4                                NaN   
..                               ...   
236                              NaN   
237     Budgetary central government   
238  Consolidated central government   
239     Budgetary central government   
240  Consolidated central government   

                IMF data dissemination standard  \
0                                           NaN   
1      General Data Dissemination System (GDDS)   
2      General Data Dissemination System (GDDS)   
3      General Data Dissemination System (GDDS)   
4                                           NaN   
..                                          ...   
236    General Data Dissemination System (GDDS)   
237    General Data Dissemination System (GDDS)   
238  Special Data Dissemination Standard (SDDS)   
239    General Data Dissemination

### Now the datatype of the columns that are float, have to be changed into object type like the rest elements

In [39]:
#problem statement to be resolved later
New_Country['National accounts reference year']=New_Country['National accounts reference year'].fillna('0').apply(lambda x: int(x) if(x!='0') else 'NaN')


ValueError: ignored

In [37]:
## Changing from float to object type


New_Country['Latest industrial data'] = New_Country['Latest industrial data'].fillna('0').apply(lambda x: int(x) if(x!='0') else 'NaN')
New_Country['Latest trade data'] = New_Country['Latest trade data'].fillna('0').apply(lambda x: int(x) if(x!='0') else 'NaN')

#Analysis of EdStatsData.csv

In [14]:
df_stats_data.head()

Country Name Country Code  \
0   Arab World          ARB   
1   Arab World          ARB   
2   Arab World          ARB   
3   Arab World          ARB   
4   Arab World          ARB   

                                      Indicator Name  Indicator Code  \
0  Adjusted net enrolment rate, lower secondary, ...      UIS.NERA.2   
1  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.F   
2  Adjusted net enrolment rate, lower secondary, ...  UIS.NERA.2.GPI   
3  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.M   
4  Adjusted net enrolment rate, primary, both sex...     SE.PRM.TENR   

        1970       1971       1972       1973       1974      1975  ...  2060  \
0        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
1        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
2        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
3        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
4  54.822121  54.894138  56.209438  57.267109  57.991138  59.36554  ...   NaN   

   2065  2070  2075  2080  2085  2090  2095  2100  Unnamed: 69  
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[5 rows x 70 columns]

In [9]:
df_stats_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 886930 entries, 0 to 886929
Data columns (total 70 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Country Name    886930 non-null  object 
 1   Country Code    886930 non-null  object 
 2   Indicator Name  886930 non-null  object 
 3   Indicator Code  886930 non-null  object 
 4   1970            72288 non-null   float64
 5   1971            35537 non-null   float64
 6   1972            35619 non-null   float64
 7   1973            35545 non-null   float64
 8   1974            35730 non-null   float64
 9   1975            87306 non-null   float64
 10  1976            37483 non-null   float64
 11  1977            37574 non-null   float64
 12  1978            37576 non-null   float64
 13  1979            36809 non-null   float64
 14  1980            89122 non-null   float64
 15  1981            38777 non-null   float64
 16  1982            37511 non-null   float64
 17  1983      

### This csv file contains unnamed: 69 that we have to remove

In [21]:
# code to remove the column unnamed: 69
New_Stats_Data = df_stats_data.drop('Unnamed: 69',axis=1)
New_Stats_Data

Country Name Country Code  \
0        Arab World          ARB   
1        Arab World          ARB   
2        Arab World          ARB   
3        Arab World          ARB   
4        Arab World          ARB   
...             ...          ...   
886925     Zimbabwe          ZWE   
886926     Zimbabwe          ZWE   
886927     Zimbabwe          ZWE   
886928     Zimbabwe          ZWE   
886929     Zimbabwe          ZWE   

                                           Indicator Name  \
0       Adjusted net enrolment rate, lower secondary, ...   
1       Adjusted net enrolment rate, lower secondary, ...   
2       Adjusted net enrolment rate, lower secondary, ...   
3       Adjusted net enrolment rate, lower secondary, ...   
4       Adjusted net enrolment rate, primary, both sex...   
...                                                   ...   
886925  Youth illiterate population, 15-24 years, male...   
886926  Youth literacy rate, population 15-24 years, b...   
886927  Youth literacy rate, population 15-24 years, f...   
886928  Youth literacy rate, population 15-24 years, g...   
886929  Youth literacy rate, population 15-24 years, m...   

              Indicator Code       1970       1971       1972       1973  \
0                 UIS.NERA.2        NaN        NaN        NaN        NaN   
1               UIS.NERA.2.F        NaN        NaN        NaN        NaN   
2             UIS.NERA.2.GPI        NaN        NaN        NaN        NaN   
3               UIS.NERA.2.M        NaN        NaN        NaN        NaN   
4                SE.PRM.TENR  54.822121  54.894138  56.209438  57.267109   
...                      ...        ...        ...        ...        ...   
886925      UIS.LP.AG15T24.M        NaN        NaN        NaN        NaN   
886926     SE.ADT.1524.LT.ZS        NaN        NaN        NaN        NaN   
886927  SE.ADT.1524.LT.FE.ZS        NaN        NaN        NaN        NaN   
886928  SE.ADT.1524.LT.FM.ZS        NaN        NaN        NaN        NaN   
886929  SE.ADT.1524.LT.MA.ZS        NaN        NaN        NaN        NaN   

             1974      1975  ...  2055  2060  2065  2070  2075  2080  2085  \
0             NaN       NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
1             NaN       NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
2             NaN       NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
3             NaN       NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
4       57.991138  59.36554  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
...           ...       ...  ...   ...   ...   ...   ...   ...   ...   ...   
886925        NaN       NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
886926        NaN       NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
886927        NaN       NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
886928        NaN       NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
886929        NaN       NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   

        2090  2095  2100  
0        NaN   NaN   NaN  
1        NaN   NaN   NaN  
2        NaN   NaN   NaN  
3        NaN   NaN   NaN  
4        NaN   NaN   NaN  
...      ...   ...   ...  
886925   NaN   NaN   NaN  
886926   NaN   NaN   NaN  
886927   NaN   NaN   NaN  
886928   NaN   NaN   NaN  
886929   NaN   NaN   NaN  

[886930 rows x 69 columns]

In [23]:
New_Stats_Data.columns

Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977', '1978',
       '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986', '1987',
       '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995', '1996',
       '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004', '2005',
       '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013', '2014',
       '2015', '2016', '2017', '2020', '2025', '2030', '2035', '2040', '2045',
       '2050', '2055', '2060', '2065', '2070', '2075', '2080', '2085', '2090',
       '2095', '2100'],
      dtype='object')

In [24]:
New_Stats_Data['Indicator Name'].nunique()

3665

In [25]:
New_Stats_Data['Country Name'].nunique()

242

### From this it can be inferred that the file contains 242 unique countries and 3665 unique indicator names corresponding to the countries

In [31]:
# Taking Descriptive Statistics of all the main columns i.e. Country name, Country code, Indicator name, Indicator code

New_Stats_Data[['Country Name','Country Code','Indicator Name','Indicator Code']].describe()



Country Name Country Code  \
count        886930       886930   
unique          242          242   
top      Arab World          ARB   
freq           3665         3665   

                                           Indicator Name Indicator Code  
count                                              886930         886930  
unique                                               3665           3665  
top     Adjusted net enrolment rate, lower secondary, ...     UIS.NERA.2  
freq                                                  242            242

# Analysis of EdStatsFootnote.csv

In [15]:
df_stats_footnote.head()

CountryCode      SeriesCode    Year          DESCRIPTION  Unnamed: 4
0         ABW  SE.PRE.ENRL.FE  YR2001  Country estimation.         NaN
1         ABW  SE.TER.TCHR.FE  YR2005  Country estimation.         NaN
2         ABW  SE.PRE.TCHR.FE  YR2000  Country estimation.         NaN
3         ABW  SE.SEC.ENRL.GC  YR2004  Country estimation.         NaN
4         ABW     SE.PRE.TCHR  YR2006  Country estimation.         NaN

In [10]:
df_stats_footnote.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 643638 entries, 0 to 643637
Data columns (total 5 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   CountryCode  643638 non-null  object 
 1   SeriesCode   643638 non-null  object 
 2   Year         643638 non-null  object 
 3   DESCRIPTION  643638 non-null  object 
 4   Unnamed: 4   0 non-null       float64
dtypes: float64(1), object(4)
memory usage: 24.6+ MB


### This csv file contains the column unnamed: 4 that have zero non-null values that have to be removed and the elements of the column "Year" contains the string YR that have to be stripped off

In [30]:
# code to remove the column unnamed: 4
New_Stats_Footnote = df_stats_footnote.drop('Unnamed: 4',axis=1)
New_Stats_Footnote['Year']=New_Stats_Footnote['Year'].str.lstrip('YR')
New_Stats_Footnote

CountryCode      SeriesCode  Year  \
0              ABW  SE.PRE.ENRL.FE  2001   
1              ABW  SE.TER.TCHR.FE  2005   
2              ABW  SE.PRE.TCHR.FE  2000   
3              ABW  SE.SEC.ENRL.GC  2004   
4              ABW     SE.PRE.TCHR  2006   
...            ...             ...   ...   
643633         ZWE     SH.DYN.MORT  2007   
643634         ZWE     SH.DYN.MORT  2014   
643635         ZWE     SH.DYN.MORT  2015   
643636         ZWE     SH.DYN.MORT  2017   
643637         ZWE     SP.POP.GROW  2017   

                                   DESCRIPTION  
0                          Country estimation.  
1                          Country estimation.  
2                          Country estimation.  
3                          Country estimation.  
4                          Country estimation.  
...                                        ...  
643633       Uncertainty bound is 91.6 - 109.3  
643634          Uncertainty bound is 54.3 - 76  
643635        Uncertainty bound is 48.3 - 73.3  
643636  5-year average value between 0s and 5s  
643637  5-year average value between 0s and 5s  

[643638 rows x 4 columns]

# Analysis of EdStatsSeries.csv

In [16]:
df_stats_series.head()

Series Code       Topic  \
0  BAR.NOED.1519.FE.ZS  Attainment   
1     BAR.NOED.1519.ZS  Attainment   
2  BAR.NOED.15UP.FE.ZS  Attainment   
3     BAR.NOED.15UP.ZS  Attainment   
4  BAR.NOED.2024.FE.ZS  Attainment   

                                      Indicator Name  \
0  Barro-Lee: Percentage of female population age...   
1  Barro-Lee: Percentage of population age 15-19 ...   
2  Barro-Lee: Percentage of female population age...   
3  Barro-Lee: Percentage of population age 15+ wi...   
4  Barro-Lee: Percentage of female population age...   

                                    Short definition  \
0  Percentage of female population age 15-19 with...   
1  Percentage of population age 15-19 with no edu...   
2  Percentage of female population age 15+ with n...   
3  Percentage of population age 15+ with no educa...   
4  Percentage of female population age 20-24 with...   

                                     Long definition  Unit of measure  \
0  Percentage of female population age 15-19 with...              NaN   
1  Percentage of population age 15-19 with no edu...              NaN   
2  Percentage of female population age 15+ with n...              NaN   
3  Percentage of population age 15+ with no educa...              NaN   
4  Percentage of female population age 20-24 with...              NaN   

  Periodicity Base Period Other notes Aggregation method  ...  \
0         NaN         NaN         NaN                NaN  ...   
1         NaN         NaN         NaN                NaN  ...   
2         NaN         NaN         NaN                NaN  ...   
3         NaN         NaN         NaN                NaN  ...   
4         NaN         NaN         NaN                NaN  ...   

  Notes from original source  General comments  \
0                        NaN               NaN   
1                        NaN               NaN   
2                        NaN               NaN   
3                        NaN               NaN   
4                        NaN               NaN   

                                              Source  \
0  Robert J. Barro and Jong-Wha Lee: http://www.b...   
1  Robert J. Barro and Jong-Wha Lee: http://www.b...   
2  Robert J. Barro and Jong-Wha Lee: http://www.b...   
3  Robert J. Barro and Jong-Wha Lee: http://www.b...   
4  Robert J. Barro and Jong-Wha Lee: http://www.b...   

  Statistical concept and methodology Development relevance  \
0                                 NaN                   NaN   
1                                 NaN                   NaN   
2                                 NaN                   NaN   
3                                 NaN                   NaN   
4                                 NaN                   NaN   

  Related source links Other web links  Related indicators  License Type  \
0                  NaN             NaN                 NaN           NaN   
1                  NaN             NaN                 NaN           NaN   
2                  NaN             NaN                 NaN           NaN   
3                  NaN             NaN                 NaN           NaN   
4                  NaN             NaN                 NaN           NaN   

   Unnamed: 20  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN  

[5 rows x 21 columns]

In [11]:
df_stats_series.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3665 entries, 0 to 3664
Data columns (total 21 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Series Code                          3665 non-null   object 
 1   Topic                                3665 non-null   object 
 2   Indicator Name                       3665 non-null   object 
 3   Short definition                     2156 non-null   object 
 4   Long definition                      3665 non-null   object 
 5   Unit of measure                      0 non-null      float64
 6   Periodicity                          99 non-null     object 
 7   Base Period                          314 non-null    object 
 8   Other notes                          552 non-null    object 
 9   Aggregation method                   47 non-null     object 
 10  Limitations and exceptions           14 non-null     object 
 11  Notes from original source    

### Dropping all the columns that have null values which are insignificant in terms of data analysis

In [29]:
# code to remove the columns that include null values
New_Stats_Series =df_stats_series.drop(['Unnamed: 20','License Type','Related indicators','Other web links','Notes from original source','Unit of measure'],axis=1)
New_Stats_Series

Series Code         Topic  \
0             BAR.NOED.1519.FE.ZS    Attainment   
1                BAR.NOED.1519.ZS    Attainment   
2             BAR.NOED.15UP.FE.ZS    Attainment   
3                BAR.NOED.15UP.ZS    Attainment   
4             BAR.NOED.2024.FE.ZS    Attainment   
...                           ...           ...   
3660    UIS.XUNIT.USCONST.3.FSGOV  Expenditures   
3661    UIS.XUNIT.USCONST.4.FSGOV  Expenditures   
3662   UIS.XUNIT.USCONST.56.FSGOV  Expenditures   
3663  XGDP.23.FSGOV.FDINSTADM.FFD  Expenditures   
3664  XGDP.56.FSGOV.FDINSTADM.FFD  Expenditures   

                                         Indicator Name  \
0     Barro-Lee: Percentage of female population age...   
1     Barro-Lee: Percentage of population age 15-19 ...   
2     Barro-Lee: Percentage of female population age...   
3     Barro-Lee: Percentage of population age 15+ wi...   
4     Barro-Lee: Percentage of female population age...   
...                                                 ...   
3660  Government expenditure per upper secondary stu...   
3661  Government expenditure per post-secondary non-...   
3662  Government expenditure per tertiary student (c...   
3663  Government expenditure in secondary institutio...   
3664  Government expenditure in tertiary institution...   

                                       Short definition  \
0     Percentage of female population age 15-19 with...   
1     Percentage of population age 15-19 with no edu...   
2     Percentage of female population age 15+ with n...   
3     Percentage of population age 15+ with no educa...   
4     Percentage of female population age 20-24 with...   
...                                                 ...   
3660                                                NaN   
3661                                                NaN   
3662                                                NaN   
3663  Total general (local, regional and central) go...   
3664  Total general (local, regional and central) go...   

                                        Long definition Periodicity  \
0     Percentage of female population age 15-19 with...         NaN   
1     Percentage of population age 15-19 with no edu...         NaN   
2     Percentage of female population age 15+ with n...         NaN   
3     Percentage of population age 15+ with no educa...         NaN   
4     Percentage of female population age 20-24 with...         NaN   
...                                                 ...         ...   
3660  Average total (current, capital and transfers)...         NaN   
3661  Average total (current, capital and transfers)...         NaN   
3662  Average total (current, capital and transfers)...         NaN   
3663  Total general (local, regional and central) go...         NaN   
3664  Total general (local, regional and central) go...         NaN   

     Base Period Other notes Aggregation method Limitations and exceptions  \
0            NaN         NaN                NaN                        NaN   
1            NaN         NaN                NaN                        NaN   
2            NaN         NaN                NaN                        NaN   
3            NaN         NaN                NaN                        NaN   
4            NaN         NaN                NaN                        NaN   
...          ...         ...                ...                        ...   
3660         NaN         NaN                NaN                        NaN   
3661         NaN         NaN                NaN                        NaN   
3662         NaN         NaN                NaN                        NaN   
3663         NaN   Secondary                NaN                        NaN   
3664         NaN    Tertiary                NaN                        NaN   

     General comments                                             Source  \
0                 NaN  Robert J. Barro and Jong-Wha Lee: http://www.b...   
1                 NaN  Robert J. Barro and Jong-W

## 3. Exploratory Analysis and Visualization
### Important Columns to analyze:
  1. 
  2.
  3.
  

## 4. F.A.Qs or doubt questions
  - 
  -
  -

## 5. Summary and Conclusion

### Insights from the visualization plots:
  - 
  -
  -

In [ ]:
# After asking and answering all the FAQs and noting down the insights into bullet points
# Finish up by writing in detailed structure.